Parser/src/main/java/ParsingTableRow.java

```diff
@@ -0,0 +1,55 @@
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class ParsingTableRow {
+    private String action;
+    private List<Symbol> symbols;
+    private List<Pair<Symbol, Integer>> goTo = new ArrayList<>();
+
+    public ParsingTableRow(List<Symbol> symbols) {
+        this.symbols = symbols;
+        for(var symbol: symbols){
+            goTo.add(new Pair<>(symbol, null));
+        }
+    }
+
+    public void setGoToSymbol(Symbol symbol, Integer state){
+        for (var pair: goTo) {
+            if(pair.getFirst().equals(symbol)){
+                pair.setSecond(state);
+            }
+        }
+    }
+
+    public Integer getGoToState(String symbol){
+        for (var s: goTo) {
```

Comment:
I am not sure what s is.

Reply:
also, above when insuccess when finding state, we return -1, here we return null => inconsistency